chapters/annotations.tex

```diff
@@ -334,15 +334,44 @@ \section{Annotations for Code Generation}\label{annotations-for-code-generation}
 The annotation \lstinline!Evaluate! can occur in the component declaration, its type declaration, or a base-class of the type-declaration.
 In the case of multiple conflicting annotations it is handled similarly to modifiers (e.g., an \lstinline!Evaluate! annotation on the component declaration takes precedence).
 In the case of hierarchical components it is applied to all components, and if both of them have annotations then \lstinline!Evaluate = true! takes precedence.
-The annotation \lstinline!Evaluate! only has effect for a component declared with the prefix \lstinline!parameter!.
+The annotation \lstinline!Evaluate! is only allowed for parameters and constants.
 
-If \lstinline!Evaluate = true!, the model developer proposes to utilize the value for the symbolic processing. In that case, it is not possible to change the parameter value after symbolic pre-processing.
+Setting \lstinline!Evaluate = true! for a structural parameter, means that it must be an evaluated parameter (i.e., its value must be determined during translation, similar to a constant).
+For a normal parameter, it has no impact, and it is recommended to issue a warning (except when the parameter is normal due to dependency on a parameter with \lstinline!Evaluate = false!, as this could be a sign of intentional overriding of \lstinline!Evaluate = true!, see example below).
+For both structural parameters and constants, the model developer further proposes to utilize the value for symbolic processing.
+A constant can never be changed after translation, and it is normal for its value to be used for symbolic processing even without \lstinline!Evaluate = true!.
 
-If \lstinline!Evaluate = false!, the model developer proposes to not utilize the value of the corresponding parameter for the symbolic processing.
+For a parameter, \lstinline!Evaluate = false! ensures that the parameter is a normal parameter according to \cref{component-variability-prefixes-discrete-parameter-constant} (meaning it is not allowed to be used where a structural expression (\cref{structural-expressions}) is expected).
```

Comment:
Quoting original comment, since there are multiple comments in the same thread:
> I feel like this would change the existing behavior. With the current phrasing, if it was required to know the value during the translation time the tool could disregard the annotation. With the proposed change the tool would have to throw an error in such a case. Is that the desirable behavior?

Yes, the proposal is written as if that is desirable.  The reasoning is that the better separation of where it is OK for parameters to be evaluated (structural expressions) and where it isn't OK (parameter expressions), we can make `Evaluate` a more predictable construct, giving more control to our users and improved portability of models between tools.

If a tool really needs to evaluate something that isn't a structural expression according to the specification, I think it would be an improvement if we could agree that evaluating nevertheless would be a tool-specific deviation form the specification.  That is, the tool should at least warn about performing this non-standard parameter evaluation.  This might be fine for a model that's only meant for use with a particular tool, but the warnings should make it clear that such a model isn't suitable for distribution as standards-compliant tool-independent Modelica code.

I should probably also mention tools with different modes of parameter evaluation, for example, allowing any `Evaluate = false` to be ignored, and force evaluation of any structural parameter.  I consider this totally fine as long as the user is aware that this doesn't correspond to standard translation, and as long as such modes aren't used to determine model validity (static checks as well as ability to produce expected simulation result).

One alternative could be to just make evaluation of normal parameters a deprecated behavior, but as I don't really see any practical difference with this approach, I prefer the simplicity of not allowing it at all according to the specification, and leave it up to tools to decide how to deal with the need to evaluate non-structural expressions.

Reply:
Now replying to the first comment:
> I still feel like the sentence on line 340 (annotations.tex) should be clarified better.
>
> 1. "For a normal parameter, it has...": Just from looking at it, "it" can refer to either the Evaluate annotation or specifically to setting Evaluate = true. Clearly, setting Evaluate = false actually has an impact as it makes the parameter normal.

The intention is that it should be clear that there's one paragraph on `Evaluate = true` and one paragraph on `Evaluate = false`.  It would be nice if this could be established somehow, so that one also doesn't need to repeat the `Evaluate = true` for the _the model developer further proposes_.  In the past, we might have used a table to make it clear, but we've lately moved away from that sort of table use.  Do you have any suggestion for how to make it clear in a good way?

> 2. The "except when..." part: I showed the sentence to an independent party not familiar with the discussion. His interpretation was that the exception is made for "has no impact" and that Evaluate = true overrides the normalcy. I believe that it was not the intent.
>
> Maybe it is better to split the sentence and state the "except when" case as a stand-alone case.

Yes, I can see the ambiguity now.  I'll split it.

Reply:
I think this can be considered resolved. I have no further questions about the change to the existing behavior, just wanted to make it explicit that it is intended. Nobody else has voiced any objections, so it's fine by me too. The comments related to phrasing have been addressed.

Reply:
OK, good!  I'll mark it resolved then, so that we can concentrate the attention of other reviewers to the unresolved conversations.